x/txfees/types/expected_keepers.go

```diff
@@ -11,16 +13,14 @@ type SpotPriceCalculator interface {
 	CalculateSpotPrice(ctx sdk.Context, poolId uint64, quoteDenom, baseDenom string) (sdk.Dec, error)
 }
 
-// GammKeeper defines the contract needed for AccountKeeper related APIs.
-type GammKeeper interface {
-	SwapExactAmountIn(
+// PoolManager defines the contract needed for swap related APIs.
+type PoolManager interface {
+	RouteExactAmountIn(
 		ctx sdk.Context,
 		sender sdk.AccAddress,
-		poolId uint64,
+		routes []swaproutertypes.SwapAmountInRoute,
 		tokenIn sdk.Coin,
-		tokenOutDenom string,
-		tokenOutMinAmount sdk.Int,
-	) (tokenOutAmount sdk.Int, err error)
+		tokenOutMinAmount sdk.Int) (tokenOutAmount sdk.Int, err error)
```

Comment:
Can we keep an API shim for single hops

Reply:
(can be done in a subsequent PR)

Reply:
Also why was this renamed to RouteExactAmountIn?

Reply:
Sounds good, will make an issue to create a separate API.

In terms of abstraction, what this method does is routes swaps across pools over several modules so the rename made sense.

I don't have a strong opinion and can change this back to multihop if you prefer.

Reply:
https://github.com/osmosis-labs/osmosis/issues/3807